### PR TITLE
cudatoolkit: generalize derivations, add cuda7

### PR DESCRIPTION
**Pull request by jb55**
There is not much different between the cudatoolkit
expressions:

```
pkgs/development/compilers/cudatoolkit/5.5.nix
pkgs/development/compilers/cudatoolkit/6.0.nix
pkgs/development/compilers/cudatoolkit/6.5.nix
```

This commit removes those and replaces them with

```
pkgs/development/compilers/cudatoolkit/generic.nix
```

and adds cudatoolkit version 7

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1172,15 +1172,28 @@ let
 
   cron = callPackage ../tools/system/cron { };
 
-  cudatoolkit5 = callPackage ../development/compilers/cudatoolkit/5.5.nix {
+  cudatoolkit5 = callPackage ../development/compilers/cudatoolkit/generic.nix {
+    version = "5.5.22";
```
Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: Remove this and add files like `v5.nix` and call `generic.nix` from them.

Comment (jb55) [inline pkgs/top-level/all-packages.nix]: Sure no problem. Just curious, but why the extra files? I noticed usage of both conventions throughout nixpkgs.

Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: Easy to find and do changes.

**Comment by jb55** [comment]
updated, let me know if that looks right

**Comment by jagajaga** [comment]
Awesome! Thank you.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1184,6 +1184,10 @@ let
     python = python26;
   };
 
+  cudatoolkit7 = callPackage ../development/compilers/cudatoolkit/7.0.nix {
```
Comment (edolstra) [inline pkgs/top-level/all-packages.nix]: Doesn't this apply `callPackage` twice?

Comment (jb55) [inline pkgs/top-level/all-packages.nix]: ah good catch, fixing

Comment (jb55) [inline pkgs/top-level/all-packages.nix]: I just copied the style from the generic ffmpeg calls, which also seem to call it twice.

**Pull request merged**